boards/arm/stm32h747i_disco/stm32h747i_disco.dtsi

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	leds {
+		compatible = "gpio-leds";
+		green_led_1:led_1 {
+			gpios = <&gpioi 12 GPIO_INT_ACTIVE_HIGH>;
+			label = "User LD1";
+		};
+		orange_led_2:led_2 {
+			gpios = <&gpioi 13 GPIO_INT_ACTIVE_HIGH>;
+			label = "User LD2";
+		};
+		red_led_3:led_3 {
+			gpios = <&gpioi 14 GPIO_INT_ACTIVE_HIGH>;
+			label = "User LD3";
+		};
+		blue_led_4:led_4 {
+			gpios = <&gpioi 15 GPIO_INT_ACTIVE_HIGH>;
+			label = "User LD4";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		wake_up: button {
+			label = "Wakeup";
+			gpios = <&gpioc 13 GPIO_INT_ACTIVE_HIGH>;
+		};
+		joy_center: joystick_center {
+			label = "joystick center";
+			gpios = <&gpiok 2 GPIO_INT_ACTIVE_LOW>;
+		};
+		joy_down: joystick_down {
+			label = "joystick down";
+			gpios = <&gpiok 3 GPIO_INT_ACTIVE_LOW>;
+		};
+		joy_up: joystick_up {
+			label = "joystick up";
+			gpios = <&gpiok 6 GPIO_INT_ACTIVE_LOW>;
+		};
+		joy_left: joystick_left {
+			label = "joystick left";
+			gpios = <&gpiok 4 GPIO_INT_ACTIVE_LOW>;
+		};
+		joy_right: joystick_right {
+			label = "joystick right";
+			gpios = <&gpiok 5 GPIO_INT_ACTIVE_LOW>;
+		};
+	};
+};
+
```

Comment:
Which core  is in charge of these IO, the M4 or the M7? i would declare it in related dtsi...
seem(IMHO) that this file should declare only peripheral shared by both the M4 and the M7 ( probably protected by HSEM protection).

Reply:
Well, this is theoretically accessible to both cores, so I found it made sense to define in a common file.
Though please note they should theoretically be accessed only though the aliases, which are split per core .dts

Reply:
This new implementation sounds better for me. stm32h747i_disco.dtsi declares resources available on boards,   stm32h747i_disco_m7.dts and stm32h747i_disco_m4.dts enable them for use.
 you can mark it as resolved